tests/components/moehlenhoff_alpha2/test_config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Test the moehlenhoff_alpha2 config flow."""
+from unittest.mock import PropertyMock, patch
+
+from moehlenhoff_alpha2 import Alpha2Base
+
+from homeassistant.components.moehlenhoff_alpha2.const import DOMAIN
+from homeassistant.helpers.typing import HomeAssistantType
+
+
+async def test_duplicate_error(hass: HomeAssistantType):
+    """Test that errors are shown when duplicates are added."""
+    Alpha2Base.name = PropertyMock(return_value="fake_base_name")
+    with patch("moehlenhoff_alpha2.Alpha2Base._fetch_static_data", return_value=True):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data={"host": "fake_host"}
+        )
+        assert result["title"] == "fake_base_name"
+        assert result["type"] == "create_entry"
+        assert not result["result"].unique_id
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data={"host": "fake_host"}
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "already_configured"
+
+
+async def test_user(hass: HomeAssistantType):
+    """Test starting a flow by user."""
+
+    with patch("moehlenhoff_alpha2.Alpha2Base._fetch_static_data", return_value=True):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}
+        )
+        assert result["type"] == "form"
+        assert result["step_id"] == "user"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={"host": "fake_host_user"}
+        )
+        assert result["type"] == "create_entry"
+        assert result["title"] == "fake_base_name"
+        assert result["data"]["host"] == "fake_host_user"
+
+
+async def test_connection_error(hass: HomeAssistantType):
+    """Test connection error."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "user"}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
should really mock the call to create the base with the error, now this will wait during the test, no need for that, mock with side-effect, like below!

Reply:
Done.